library/psa_crypto.c

```diff
@@ -545,6 +545,68 @@ static inline size_t psa_get_key_slot_bits( const psa_key_slot_t *slot )
     return( slot->attr.bits );
 }
 
+/** Return the output MAC length of a MAC algorithm, in bytes
+ *
+ * \param[in] algorithm     The specific MAC algorithm
+ * \param[in] key_type      The key type of the key to be used with the
+ *                          \p algorithm.
+ * \param[out] length       The calculated output length of the given MAC
+ *                          \p algorithm when used with a key corresponding to
+ *                          the given \p key_type
+ *
+ * \retval #PSA_SUCCESS
+ *         The \p length has been successfully calculated
+ * \retval #PSA_ERROR_NOT_SUPPORTED
+ *         \p algorithm is a MAC algorithm, but ubsupported by this PSA core.
+ * \retval #PSA_ERROR_INVALID_ARGUMENT
+ *         \p algorithm is not a valid, specific MAC algorithm or \p key_type
+ *         describes a key incompatible with the specified \p algorithm.
+ */
+static psa_status_t psa_get_mac_output_length( psa_algorithm_t algorithm,
+                                               psa_key_type_t key_type,
+                                               size_t *length )
+{
+    if( !PSA_ALG_IS_MAC( algorithm ) || PSA_ALG_IS_WILDCARD( algorithm ) )
+        return( PSA_ERROR_INVALID_ARGUMENT );
+
+    size_t default_length = 0;
+
+    if( PSA_ALG_FULL_LENGTH_MAC( algorithm ) == PSA_ALG_CMAC ||
```

Comment:
Each if block should be guarded by the relevant `#if PSA_WANT_xxx`.

Reply:
There's no `PSA_WANT_xxx` defined for CMAC (nor CBC-MAC) yet. Additionally, I'm unsure of the value of guarding here, besides the very slightly reduced code size. All operations inside this function use symbols which are unconditionally available.

Reply:
After reworking and relying on `PSA_MAC_LENGTH` instead, there is nothing left to guard.